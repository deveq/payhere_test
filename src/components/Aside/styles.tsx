import styled from 'styled-components';
import {flexCenter} from 'styles/helper';

export const Aside = styled.aside`
    width: 200px;
    background: ${({theme}) => theme.bg_page1};
    padding: 30px;
    box-sizing: border-box;
    flex-shrink: 0;
`;

export const List = styled.ul`
    display: grid;
    gap: 5px;
`;

export const Title = styled.h2`
    font-size: 24px;
    font-weight: bold;
    border-bottom: ${({theme}) => `1px solid ${theme.bg_element4}`};
    color: ${({theme}) => theme.text1};
    padding-bottom: 10px;
    margin-bottom: 20px;
    text-indent: 10px;
`;

export const ListItem = styled.li<{selected: boolean}>`
    background: ${({selected,theme}) => selected ? theme.bg_element1 : 'transparent'};
    padding: 15px 20px;
    border-radius: 10px;
    font-weight: bold;
    color: ${({selected, theme}) => selected ? theme.text2 : theme.text3};
    &:hover,
    &:active {
        background: ${({theme}) => theme.bg_element1};
    };
    cursor: pointer;
`;

export const Subtitle = styled.div`

`;