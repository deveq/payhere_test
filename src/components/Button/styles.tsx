import styled from "styled-components";

type ButtonSize = 'sm' | 'md' | 'lg';
export interface ButtonProps {
    size?: ButtonSize;
    backgroundColor?: string;
    borderColor?: string;
    color?: string;
    showShadow?: boolean;
}

const handleSize = (size?: ButtonSize) => {
    if (size === 'sm') return '0.5rem 1rem';
    if (size === 'md') return '0.75rem 1.5rem';
    if (size === 'lg') return '1rem 2rem';
    return '0.75rem 1.5rem';
}

export const Button = styled.button<ButtonProps>`
    background: ${({backgroundColor, theme}) => 
        backgroundColor || theme.bg_element2
    };
    border-radius: 5px;
    color: ${({color, theme}) => 
        color || theme.text1
    };
    padding: ${({size}) => handleSize(size)};
    font-weight: bold;
    box-shadow: ${({theme, showShadow}) => showShadow ? theme.boxShadow : 'none'};
    border: ${({showShadow, theme}) => showShadow ? 'none' : `1px solid ${theme.border2}` };
    &:hover,
    &:active {
        background: ${({theme}) => theme.bg_element3};
    }
    cursor: pointer;
`;

Button.defaultProps = {
    size: 'md',
}

export const Primary = styled(Button)`
    background: ${({theme}) => theme.primary};
    color: white;
`;

export const Destructive = styled(Button)`
    background: ${({theme}) => theme.destructive};
    color: white;
    border: none;
    &:hover,
    &:active {
        background: ${({theme}) => theme.destructive};
    }
`;