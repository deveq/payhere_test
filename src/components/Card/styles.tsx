import styled from 'styled-components';

export const CardContainer = styled.div`
    background: ${({theme}) => theme.bg_element1};
    border-radius: 10px;
    padding: 5px 10px;
`;