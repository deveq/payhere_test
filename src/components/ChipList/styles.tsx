import styled from 'styled-components';

interface ChipProps {
    selected?: boolean;
}

export const Chip = styled.div<ChipProps>`
    padding: 5px 10px;
    max-width: 150px;
    min-width: 40px;
    text-align: center;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    border-radius: 20px;
    border: 1px solid ${({theme}) => theme.border2};
    background: ${({selected, theme}) => selected ? theme.primary : 'white'};
    color: ${({selected}) => selected ? 'white' : '#333'};
    cursor: pointer;
    &:active {
        opacity: 0.9;
    }
`;

export const ChipList = styled.div`
    display: flex;
    /* flex-direction: column;
    align-items: flex-start; */
    flex-wrap: wrap;
    gap: 8px;
    margin-bottom: 10px;
`;

export const Message = styled.p`
    color: ${({theme}) => theme.text1};
`;