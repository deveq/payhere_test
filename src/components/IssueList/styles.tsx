import styled from 'styled-components';
import {BsCheckCircle, BsSlashCircle, } from 'react-icons/bs';
import {TbCircleDot} from 'react-icons/tb';
import {IconBaseProps} from 'react-icons'

export type IssueType = 'open' | 'closed' | 'notPlanned';

interface StatusProps extends IconBaseProps {
    status: IssueType;
}

export const Container = styled.li`
    display: flex;
    background: ${({theme}) => theme.bg_page2};
    border: 1px solid ${({theme}) => theme.border3};
    border-radius: 10px;
    padding: 20px 10px;
`;

export const Body = styled.div`
    display: flex;
    flex-direction: column;
    gap: 5px;
`;

export const RepositoryInfo = styled.div`
    color: ${({theme}) => theme.text2};
    font-weight: bold;
    font-size: 14px;
`;

export const Title = styled.div`
    color: ${({theme}) => theme.primary};
    font-weight: bold;
    white-space: nowrap;
    text-overflow: ellipsis;
    cursor: pointer;
`;

export const Context = styled.p`
    color: ${({theme}) => theme.text1};
    font-size: 14px;
    max-height: 40px;
    overflow: hidden;
    text-overflow: ellipsis;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
`;

export const Info = styled.div`
    color: ${({theme}) => theme.text2};
    font-size: 12px;
`;

const StyledStatus = ({status, ...rest}: StatusProps) => {
    if (status === 'open') {
        return <TbCircleDot {...rest} color="#3fb950" />
    } else if (status === 'closed') {
        return <BsCheckCircle {...rest} color="#a371f7" />
    } else if (status === 'notPlanned') {
        return <BsSlashCircle {...rest} color="#8b9491" />
    }
    return <TbCircleDot {...rest} />
}

export const Status = styled(StyledStatus)`
    width: 25px;
    height: 25px;
    flex-shrink: 0;
    margin-right: 5px;
`;

export const IssueList = styled.ul`
    display: flex;
    flex-direction: column;
    gap: 10px;
    flex-shrink: 0;
    max-height: 75vh;
    overflow: auto;
`;