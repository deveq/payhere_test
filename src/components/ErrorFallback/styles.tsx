import styled from 'styled-components';
import {flexCenter} from 'styles/helper';
import {FaRegSadTear} from 'react-icons/fa';

export const Container = styled.div`
    height: 100vh;
    ${flexCenter};
    flex-direction: column;
`;

export const Paragraph = styled.p`
    font-size: 30px;
`;

export const ErrorMessage = styled.pre`
    font-size: 20px;
`;

export const SadIcon = styled(FaRegSadTear)`
    width: 100px;
    height: 100px;
    color: gray;
`;