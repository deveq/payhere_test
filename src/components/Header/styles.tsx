import React from 'react';
import styled from 'styled-components';
import { ReactComponent } from 'assets/logo.svg';
import {flexCenter} from 'styles/helper';


export const Container = styled.header`
	display: flex;
	padding: 1rem 2rem;
	background: ${(props) => props.theme.bg_page1};
	box-shadow: ${(props) => props.theme.boxShadow};
`;

export const Logo = styled(ReactComponent)`
	width: 50px;
	height: 50px;
	${flexCenter};
`;

export const Headline = styled.h1`
`;

export const Nav = styled.nav`
	flex: 1;
	${flexCenter};
`;

export const LoginContainer = styled.div`

`;