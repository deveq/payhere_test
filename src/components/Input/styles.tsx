import { Children } from 'react';
import styled from 'styled-components';
import { FiSearch } from 'react-icons/fi';
import { flexCenter } from 'styles/helper';

interface HistoryProps {
	show?: boolean;
}

export const Container = styled.form`
	${flexCenter};
	width: 250px;
	border-radius: 10px;
	margin-left: 20px;
	padding: 10px;
	border: 1px solid ${({ theme }) => theme.border3};
	background: ${({ theme }) => theme.bg_element1};
	position: relative;
`;

export const Input = styled.input`
	border: none;
	outline: none;
	flex: 1;
	font-size: 16px;
	color: ${({ theme }) => theme.text1};
	background: ${({ theme }) => theme.bg_element1};
`;

export const Icon = styled(FiSearch).attrs({
	'data-testid': 'icon'
})`
	width: 20px;
	height: 20px;
	color: ${({ theme }) => theme.text1};
	cursor: pointer;
	&:hover,
	&:active {
		opacity: 0.6;
	}
`;

export const History = styled.ul<HistoryProps>`
	display: ${({ show }) => (show ? 'flex' : 'none')};
	position: absolute;
	top: 45px;
	background: ${({ theme }) => theme.bg_element1};
	width: 100%;
	height: ${({ children }) => `${Children.count(children) * 40}`}px;
	border-radius: 10px;
	z-index: 10;
	box-shadow: ${({ theme }) => theme.boxShadow};
	padding: 10px 20px;
	box-sizing: border-box;
	flex-direction: column;
	justify-content: space-around;
`;

export const HistoryItem = styled.li`
	color: ${({ theme }) => theme.text1};
	font-size: 18px;
	display: flex;
	justify-content: space-between;
	max-width: 90%;
	white-space: nowrap;
	text-overflow: ellipsis;
	overflow: hidden;
	&:hover {
		opacity: 0.6;
	}
	cursor: pointer;
`;
