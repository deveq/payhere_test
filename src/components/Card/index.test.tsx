import {screen} from '@testing-library/react';
import {renderWithWrapper} from 'tests';
import Card from './index';
import {lightTheme} from 'styles/theme';

describe('<Card />', () => {
    const MockEl = () => (
        <div data-testid="mock"/>
    )

    it('정상적으로 렌더링되는지 확인', () => {
        const {container} = renderWithWrapper(<Card>
            <MockEl />
        </Card>);

        const mockEl = screen.getByTestId('mock');
        const card = mockEl.parentElement;
        expect(card).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it('정상적으로 렌더링되는지 확인', () => {
        const {container} = renderWithWrapper(<Card>
            <MockEl />
        </Card>);

        const mockEl = screen.getByTestId('mock');
        const card = mockEl.parentElement;
        expect(card).toHaveStyle({
            background: lightTheme.bg_element1,
            'border-radius': '10px',
            padding: '5px 10px',
        })
    })
})