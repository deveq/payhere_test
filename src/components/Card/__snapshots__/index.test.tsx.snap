// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> 정상적으로 렌더링되는지 확인 1`] = `
<div>
  <div
    class="sc-bcXHqe jcLPlY"
  >
    <div
      data-testid="mock"
    />
  </div>
</div>
`;
